apps/src/componentLibrary/slider/Slider.tsx

```diff
@@ -52,29 +101,223 @@ const Slider: React.FunctionComponent<SliderProps> = ({
+
+  const [backgroundStyle, setBackgroundStyle] = useState('');
+
+  // Override min and max values for percent mode
+  const minSliderValue = isPercentMode && minValue === undefined ? 0 : minValue;
+  const maxSliderValue =
+    isPercentMode && minValue === undefined ? 100 : maxValue;
+
+  // TODO: Uncomment when working on adding steps support OR working on adding center mark
+  // // Calculate the center value based on min and max values
+  // const centerValue = (Number(minValue) + Number(maxValue)) / 2;
+
+  // Calculate percentage fill for gradient
+  const calculateFillPercent = useCallback(
+    (value: number) => {
+      const center = (Number(minSliderValue) + Number(maxSliderValue)) / 2;
+      if (isCentered) {
+        if (value >= center) {
+          // Value is on the right side of the center
+          return {
+            fillPercent:
+              ((value - center) / (Number(maxSliderValue) - center)) * 50 + 50,
+            leftFill: false, // No left fill, fill right side only
+          };
+        } else {
+          // Value is on the left side of the center
+          return {
+            fillPercent:
+              50 - ((center - value) / (center - Number(minSliderValue))) * 50,
+            leftFill: true, // Left fill from center
+          };
+        }
+      } else {
+        // Regular fill calculation
+        return {
+          fillPercent:
+            (100 * (value - Number(minSliderValue))) /
+            (Number(maxSliderValue) - Number(minSliderValue)),
+          leftFill: false,
+        };
+      }
+    },
+    [isCentered, minSliderValue, maxSliderValue]
+  );
+
+  // TODO: Uncomment when working on adding steps support
+  // // Function to snap the value to the nearest step in the steps array
+  // const snapToStep = (value: number) => {
+  //   if (!steps || steps.length === 0) {
+  //     return value; // If no steps are provided, return the value as is
+  //   }
+  //   return steps.reduce((prev, curr) =>
+  //     Math.abs(curr - value) < Math.abs(prev - value) ? curr : prev
+  //   );
+  // };
+
+  // Update the value on input change and snap it to the nearest step
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    // TODO: Uncomment when working on adding steps support
+    // const snappedValue = snapToStep(Number(event.target.value));
+
+    const snappedValue = Number(event.target.value);
+    onChange({...event, target: {...event.target, value: `${snappedValue}`}});
+  };
+
+  const handleControlButtonClick = useCallback(
+    (operation: 'subtract' | 'add') => {
+      const newValue = operation === 'subtract' ? +value - step : +value + step;
+
+      if (newValue >= minValue && newValue <= maxValue) {
+        onChange({
+          target: {value: `${newValue}`},
+        } as ChangeEvent<HTMLInputElement>);
+      }
+    },
+    [maxValue, minValue, onChange, step, value]
+  );
+
+  // TODO: Uncomment when working on adding steps support
```

Comment:
Out of curiosity, what's the plan for this feature?

Reply:
There's a plan to allow consumers to pass a set of steps(values) that can be selected in specified range and user will only be able to select only from these steps.
e.g. time range with steps [0 min, 15 min, 30min, 60 min, 95+ min]